src/pages/Opening/Opening.jsx

```diff
@@ -0,0 +1,246 @@
+import { gql } from 'apollo-boost';
+import React, { Component, Fragment } from 'react';
+import { Mutation, Query } from 'react-apollo';
+import { Button, DataTable, TableHeader, TableBody, TableRow, TableColumn, Snackbar, MenuButtonColumn, FontIcon } from 'react-md';
+import { withRouter } from 'react-router-dom';
+import OpeningDialog from './Dialog';
+import Loading from '../../components/Loading';
+import NumberFormat from 'react-number-format';
+import ConfirmDialog from '../../components/ConfirmDialog';
+
+class Opening extends Component {
+  state = {
+    showOpeningDialog: false,
+    openingToDelete: null,
+    alertVisible: false,
+    toasts: []
+  };
+
+  emptyOpening = {
+    id: null,
+    jobTitle: '',
+    jobDescription: '',
+    company: '',
+    maxSalaryRange: '',
+    status: '',
+    steps: [],
+    applications: []
+  };
+
+  opening = this.emptyOpening;
+  headers = ['Job Title', 'Job Description', 'Company', 'Max Salary Range', 'status'];
+  menuItems = [
+    {
+      key: 'edit',
+      leftIcon: <FontIcon>edit</FontIcon>,
+      primaryText: 'Edit'
+    },
+    { divider: true },
+    {
+      key: 'delete',
+      leftIcon: <FontIcon className="md-text--error">delete</FontIcon>,
+      primaryText: <span className="md-text--error">Delete</span>
+    }
+  ];
+
+  showDialog = () => {
+    this.setState({ showOpeningDialog: true });
+  };
+
+  hideDialog = () => {
+    this.setState({ showOpeningDialog: false });
+    this.opening = this.emptyOpening;
+  };
+
+  onClickEdit = openingToEdit => {
+    openingToEdit.steps = openingToEdit.steps.map(step => step.id);
```

Comment:
I don't know why, but when clicking _Edit_ to an opening the third time doesn't returns its steps nor the applications (the lists are empty) and if we click it one more time (4th) returns the error `TypeError: Cannot read property 'id' of undefined` in this line

Reply:
Fixed. Thanks for testing